api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/constraint/ListOfDatesInPastValidator.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.dhsc.htbhf.claimant.model.constraint;
+
+import org.springframework.util.CollectionUtils;
+
+import java.time.LocalDate;
+import java.util.List;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class ListOfDatesInPastValidator implements ConstraintValidator<ListOfDatesInPast, List<LocalDate>> {
+
+    @Override
+    public boolean isValid(List<LocalDate> dates, ConstraintValidatorContext context) {
+        if (CollectionUtils.isEmpty(dates)) {
+            return true;
+        }
+        return allDatesInPast(dates);
+    }
+
+    private boolean allDatesInPast(List<LocalDate> dates) {
+        LocalDate now = LocalDate.now();
```

Comment:
For loop and below can be replaced with a one liner `return dates.stream().noneMatch(now::isBefore);`

Reply:
I like the simplicity of my version, so I'm going to keep it, this one makes me think too much for something that is very simple.